lib/presentation/widget/home/home_widget.dart

```diff
@@ -58,21 +61,35 @@ class _HomeWidgetState extends State<HomeWidget> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      key: _scaffoldKey,
-      appBar: AppBar(
-        automaticallyImplyLeading: false,
-        title: Text(
-            AppLocalizations.of(context).my_space_title,
-            style: TextStyle(fontSize: 24, color: Colors.white)),
-        centerTitle: true,
-        backgroundColor: AppColor.primaryColor,
-        leading: IconButton(
-            icon: SvgPicture.asset(imagePath.icLinShareMenu),
-            onPressed: () => _scaffoldKey.currentState.openDrawer()),
-      ),
-      drawer: SideMenuDrawerWidget(),
-      body: getIt<MySpaceWidget>(),
+    return StoreConnector(
```

Comment:
We should not wrap hold widget by StoreConnector. Because when ever AppState is changed, whole Widget is rebuilded. For example:
The State has 10 sub-state. The widget just use 1 state. Then when any of 9 remain sub-state is changed, our widget also will be rebuilded.

For example code:
```
....
      appBar: AppBar(
        automaticallyImplyLeading: false,
        title: StoreConnector<AppState, String>(
          converter: (store) => store.state.uiState.currentView.name,
          distinct: true,
          builder: (context, screenName) {
            return Text(screenName,
                style: TextStyle(fontSize: 24, color: Colors.white));
          },
        ),
        centerTitle: true,
        backgroundColor: AppColor.primaryColor,
        leading: IconButton(
            icon: SvgPicture.asset(imagePath.icLinShareMenu),
            onPressed: () => _scaffoldKey.currentState.openDrawer()),
      ),
....
```
The way `StoreConnector` works is when app state changed, it will call converter and pick a `...currentView.name` to check before decide to call builder. About the check for rebuild,  when `distinct` is enable, `StoreConnector` will compare the last `...currentView.name` value with the current `...currentView.name` value is same or not, if two values are the same, rebuild won't be called, and otherwise. 

Therefore, the widget Text ONLY rebuild when `...currentView.name` is updated in AppState. It will save our app performance.

Reply:
The idea same with `setState`.
https://flutter.dev/docs/perf/rendering/best-practices#controlling-build-cost

Reply:
You are right, I forgot this :(

Reply:
So we can only listen to the whole state ? I mean this will be tough when we'll have a lot of states

Reply:
pls check again shared space and home widget